pkg/bootstrap/docker/dockerhelper/client.go

```diff
@@ -0,0 +1,285 @@
+package dockerhelper
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"time"
+
+	"golang.org/x/net/context"
+
+	"github.com/docker/docker/pkg/stdcopy"
+	"github.com/docker/engine-api/client"
+	"github.com/docker/engine-api/types"
+	"github.com/docker/engine-api/types/container"
+	"github.com/docker/engine-api/types/network"
+)
+
+const (
+	// defaultDockerOpTimeout is the default timeout of short running docker operations.
+	defaultDockerOpTimeout = 10 * time.Minute
+)
+
+// NewClient creates an instance of the client Interface, given a docker engine
+// client
+func NewClient(endpoint string, client *client.Client) Interface {
+	return &dockerClient{
+		endpoint: endpoint,
+		client:   client,
+	}
+}
+
+type dockerClient struct {
+	endpoint string
```

Comment:
the link didn't work for me - I can see it's used in Endpoint(), but I can't see Endpoint() being used anywhere...